### PR TITLE
[ZEPPELIN-820] Reduce websocket communication by unicasting instead of broadcasting note list 

### PR DESCRIPTION
**Pull request by prasadwagle**
### What is this PR for?
Reducing websocket communication by unicasting instead of broadcasting notes list reduces probability of deadlock in jetty8 and also improves response time.

### What type of PR is it?
Improvement

### What is the Jira issue?
[ZEPPELIN-820] (https://issues.apache.org/jira/browse/ZEPPELIN-820)

### How should this be tested?
1. Open two browser windows.
2. Check if note list shows up on home page.
3. Check if creating, removing note in one window results in updates to notes list in other windows.

### CODE REVIEW

**Comment by bzz** [comment]
Test failure looks quite strange

```
[INFO] Building Zeppelin: Server 0.6.0-incubating-SNAPSHOT
...
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.apache.zeppelin.rest.ZeppelinSparkClusterTest
....
01:13:27,991  INFO org.apache.zeppelin.socket.NotebookServer:1110 - Job 20160422-011317_510294549 is finished
01:13:28,015  INFO org.apache.zeppelin.notebook.Paragraph:235 - run paragraph 20160422-011328_1891493426 using pyspark org.apache.zeppelin.interpreter.LazyOpenInterpreter@2373e70d

No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.

The build has been terminated
```

@prasadwagle could you please close and re-opening the PR again to trigger a CI and see if this deserves a JIRA issue with `flaky-test` label?

**Comment by prasadwagle** [comment]
@bzz New build succeeded. Created [ZEPPELIN-822](https://issues.apache.org/jira/browse/ZEPPELIN-822) - Flaky test - org.apache.zeppelin.rest.ZeppelinSparkClusterTest

File: zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java [hunk 1]
```diff
@@ -339,6 +338,14 @@ private void broadcastAll(Message m) {
     }
   }
 
+  private void unicast(Message m, NotebookSocket conn) {
+    try {
+      conn.send(serializeMessage(m));
+    } catch (IOException e) {
+      LOG.error("socket error", e);
```
Comment (felixcheung) [inline zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java]: is it potentially a problem to catch and ignore an error with this?

Comment (prasadwagle) [inline zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java]: @felixcheung I am using the approach used in the functions below:
  private void broadcast(String noteId, Message m) {
  private void broadcastExcept(String noteId, Message m, NotebookSocket exclude) {

**Comment by Leemoonsoo** [comment]
This PR does not update note title in the menu and main page, when title is changed.
Try change note title and see list of note in the menu and main page.

I think broadcastNoteList need to be called when note title is updated.

**Comment by Leemoonsoo** [comment]
LGTM

**Comment by Leemoonsoo** [comment]
Merge if there're no more discussions

**Pull request closed without merging**